README.md

```diff
@@ -28,65 +50,95 @@ cd ndx-miniscope
 pip install -e .
 ```
 
-### Usage
+The following code demonstrates the usage of this extension to convert Miniscope acquisition data into NWB.
 
+### Usage
 
 ```python
-import os
-from ndx_miniscope import read_settings, read_notes, read_miniscope_timestamps, get_starting_frames
-from pynwb import NWBFile, NWBHDF5IO
 from datetime import datetime
 from dateutil.tz import tzlocal
+import glob
+import os
+from pynwb import NWBFile, NWBHDF5IO
 from pynwb.image import ImageSeries
 from natsort import natsorted
-from glob import glob
 
+from ndx_miniscope.utils import (
+    add_miniscope_device,
+    get_starting_frames,
+    get_timestamps,
+    read_miniscope_config,
+    read_notes,
+)
 
-data_dir = 'path/to/data_dir'
+# The main folder that contains subfolders with the Miniscope data
+folder_path = "C6-J588_Disc5/"
 
+# Create the NWBFile
 session_start_time = datetime(2017, 4, 15, 12, tzinfo=tzlocal())
+nwbfile = NWBFile(
+    session_description="session_description",
+    identifier="identifier",
+    session_start_time=session_start_time,
+)
 
-nwb = NWBFile('session_description', 'identifier', session_start_time)
+# Load the miscroscope settings
+miniscope_folder_path = "C6-J588_Disc5/15_03_28/Miniscope/"
+miniscope_metadata = read_miniscope_config(folder_path=miniscope_folder_path)
+# Create the Miniscope device with the microscope metadata and add it to NWB
+add_miniscope_device(nwbfile=nwbfile, device_metadata=miniscope_metadata)
 
-miniscope = read_settings(data_dir)
-nwb.add_device(miniscope)
+# Load the behavioral camera settings
+behavcam_folder_path = "C6-J588_Disc5/15_03_28/BehavCam_2/"
+behavcam_metadata = read_miniscope_config(folder_path=behavcam_folder_path)
+# Create the Miniscope device with the behavioral camera metadata and add it to NWB
+add_miniscope_device(nwbfile=nwbfile, device_metadata=behavcam_metadata)
 
-annotations = read_notes(data_dir)
-if annotations is not None:
-    nwb.add_acquisition(annotations)
+# Loading the timestamps
+behavcam_timestamps = get_timestamps(folder_path=folder_path, file_pattern="BehavCam*/timeStamps.csv")
+# Load the starting frames of the video files
+# Note this function requires to have `cv2` installed
+starting_frames = get_starting_frames(folder_path=folder_path, video_file_pattern="*/BehavCam*/*.avi")
 
-ms_files = natsorted(glob(os.path.join(data_dir, 'msCam*.avi')))
-behav_files = natsorted(glob(os.path.join(data_dir, 'behavCam*.avi')))
+# TODO: redirect to MiniscopeConverter
```

Comment:
Basically this part should be about how to add the `OnePhotonSeries` and `ImageSeries` to nwb which could be a redirect to the converter in `neuroconv`?

Reply:
Yeah, that sounds good for the README!